Мамонтов Дмитрий, 20-я когорта. проект 7 спринта.

DmitriyMamontov opened the pull request:
Первая версия

DmitriyMamontov committed:
Мамонтов Дмитрий, 20-я когорта. проект 7 спринта.
Первая версия

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,68 @@
+    }
+
+    cleanup_registered_couriers.append(courier_info)
+    yield courier_info
```
Можно лучше: если в фикстуре после yield ничего не выполняется, то правильнее будет использовать return

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,68 @@
+def order_data():
+    created_tracks = []
+
+    def create(color=None):
```
Можно лучше: не стоит определять функции внутри фикстур. Задачи фикстур выполняться до тестового метода (выполнять предусловия). Определяя внутри фикстуры функцию, появляться необходимость вызывать эту функцию внутри тестового метода.

ShamilovR commented on test_create_courier.py:
```diff
@@ -0,0 +1,62 @@
+        with allure.step("Проверка статус кода"):
+            assert duplicate_response.status_code == 409
+        with allure.step("Проверка сообщения в ответе"):
+            assert duplicate_response.json()["message"] == "Этот логин уже используется. Попробуйте другой."
```
Можно лучше: здесь и далее: тексты ответов от сервера следует выносить в дата модули. В случае изменении текста ответа от сервера, не придется править в каждом тесте отдельно. Достаточно будет поправить лишь в одном месте - дата модуле.

ShamilovR commented on test_create_courier.py:
```diff
@@ -0,0 +1,62 @@
+import allure
+
+class TestCourierCreation:
+
```
Нужно исправить: тестовые модули должны храниться в директории tests

DmitriyMamontov committed:
Мамонтов Дмитрий, 20-я когорта. проект 7 спринта.
Вторая версия, исправлено обязательное замечание, и два необязательных. 1 необязательное не исправил из-за сроков

ShamilovR commented on conftest.py:
```diff
@@ -0,0 +1,69 @@
+from courier_methods import CourierMethods
+from order_methods import OrderMethods
+
+fake = Faker()
+created_couriers = []
+
```
Можно лучше: модуль conftest предназначен только для фикстур и хуков.
